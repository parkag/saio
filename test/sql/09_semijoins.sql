BEGIN;
CREATE TABLE table_0(
    col_0 TEXT,
    col_1 TEXT
);
CREATE TABLE table_1(
    col_0 TEXT,
    col_1 TEXT,
    col_2 TEXT
);
CREATE TABLE table_2(
    col_0 TEXT,
    col_1 TEXT,
    col_2 TEXT
);

LOAD 'saio';
SET saio_threshold to 2;

EXPLAIN (COSTS OFF) SELECT * FROM table_0
RIGHT JOIN table_1 ON (table_0.col_0 = table_1.col_1)
RIGHT JOIN table_2 ON (table_0.col_0 = table_2.col_1)
WHERE table_1.col_2 in (SELECT col_1 FROM table_2);

rollback;
