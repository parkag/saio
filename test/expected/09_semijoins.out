BEGIN;
CREATE TABLE table_0(
    col_0 TEXT,
    col_1 TEXT
);
CREATE TABLE table_1(
    col_0 TEXT,
    col_1 TEXT,
    col_2 TEXT
);
CREATE TABLE table_2(
    col_0 TEXT,
    col_1 TEXT,
    col_2 TEXT
);
LOAD 'saio';
SET saio_threshold to 2;
EXPLAIN (COSTS OFF) SELECT * FROM table_0
RIGHT JOIN table_1 ON (table_0.col_0 = table_1.col_1)
RIGHT JOIN table_2 ON (table_0.col_0 = table_2.col_1)
WHERE table_1.col_2 in (SELECT col_1 FROM table_2);
                            QUERY PLAN                             
-------------------------------------------------------------------
 Merge Join
   Merge Cond: (table_1.col_1 = table_0.col_0)
   ->  Merge Join
         Merge Cond: (table_1.col_1 = table_2.col_1)
         ->  Sort
               Sort Key: table_1.col_1
               ->  Hash Join
                     Hash Cond: (table_1.col_2 = table_2_1.col_1)
                     ->  Seq Scan on table_1
                     ->  Hash
                           ->  HashAggregate
                                 Group Key: table_2_1.col_1
                                 ->  Seq Scan on table_2 table_2_1
         ->  Sort
               Sort Key: table_2.col_1
               ->  Seq Scan on table_2
   ->  Sort
         Sort Key: table_0.col_0
         ->  Seq Scan on table_0
(19 rows)

rollback;
